tutorials/cloudxr_runtime_for_xr_applications/README.md

```diff
@@ -0,0 +1,143 @@
+
+**What you'll accomplish:**
+- Set up CloudXR Runtime as a bridge between Holoscan applications and XR devices
+- Configure Apple Vision Pro as an XR client
+- Successfully stream and experience immersive 3D content from XR applications to Apple Vision Pro
+
+<p align="center">
+  <img src="../../applications/xr_gsplat/doc/gsplat-demo.gif" alt="Gsplat Demo on AVP">
+</p>
+
+## Table of Contents
+- [System Requirements](#system-requirements)
+- [Run CloudXR Runtime Container with Docker](#run-cloudxr-runtime-container-with-docker)
+- [Run Your XR Application](#run-your-xr-application)
+- [Set up Apple Vision Pro](#set-up-apple-vision-pro)
+- [Troubleshooting](#troubleshooting)
+
+## System Requirements
+
+### Apple Vision Pro
+- visionOS 2.0 or later
+- Apple M3 Pro chip with an 11-core CPU (minimum 5 performance cores and 6 efficiency cores)
+- 16GB unified memory
+- 256GB SSD storage
+
+### Apple Silicon Mac
+Required for building the Isaac XR Teleop Sample Client App for Apple Vision Pro with Xcode:
+- macOS Sonoma 14.5 or later
+- Xcode development environment
+
+## Run CloudXR Runtime Container with Docker
+
+The CloudXR Runtime runs in a Docker container on the same machine as your application, serving as a bidirectional bridge that streams sensor data from the XR device to your application and display content back to the XR device.
+
+### Step 1: Configure Firewall
+Ensure your firewall allows connections to the ports used by CloudXR:
+
+```bash
+sudo ufw allow 47998:48000,48005,48008,48012/udp
+sudo ufw allow 48010/tcp
+```
+
+### Step 2: Create Shared Directory
+Create a shared directory for communication between XR applications and the CloudXR Runtime:
+
+```bash
+mkdir -p $(pwd)/openxr
+```
+
+This directory will be used for temporary cache files and runtime communication.
+
+### Step 3: Start CloudXR Runtime Container
+Launch the CloudXR Runtime container, mounting the shared directory:
+
+```bash
+docker run -it --rm --name cloudxr-runtime \
+    --user $(id -u):$(id -g) \
+    --gpus=all \
+    -e "ACCEPT_EULA=Y" \
+    --mount type=bind,src=$(pwd)/openxr,dst=/openxr \
+    -p 48010:48010 \
+    -p 47998:47998/udp \
+    -p 47999:47999/udp \
+    -p 48000:48000/udp \
+    -p 48005:48005/udp \
+    -p 48008:48008/udp \
+    -p 48012:48012/udp \
+    nvcr.io/nvidia/cloudxr-runtime:0.1.0-isaac
+```
+
+The container should start and wait for client connections.
+
+## Run Your XR Application
+This tutorial can work across XR applications in holohub, including [volume_rendering_xr](../../applications/volume_rendering_xr/), [xr_gsplat](../../applications/xr_gsplat/), and [xr_holoviz](../../applications/xr_holoviz/).
+
+### Step 1: Build XR Application
+By default, you can build your chosen Holoscan XR application using the command:
+```shell
+# from the root directory of holohub
+./holohub run-container <app-name>
```

Comment:
Let's mention that `holohub run-container` mounts the HoloHub directory by default, which is why the OpenXR sockets from `holohub/openxr` on the host are available in the app container